lib/routing/router.ts

```diff
@@ -22,14 +22,14 @@ export class Router<M extends Middleware, C extends Controller, R extends Route<
   ) { }
 
   public route(routes: R[]): void {
-    this.requestPath = this.request.getPath()
-    this.requestMethod = this.request.getRequestMethod();
+    this.requestPath = this.request.path;
+    this.requestMethod = this.request.method;
 
     try {
       this.subjectRoute = this.getRequestRoute(routes);
     } catch (e) {
       throw e; // could not find route, lets just throw for now.
-    };
+    }
 
     // add path params to request object
     Object.keys(this.pathParams).forEach(param => {
```

Comment:
I think the `this.request.add(param, this.pathParams[param]);` in the route function should be `this.request.add(param, this.pathParams[param],true);` . This will allow people to set data prior to routing and then the path params will just override that

Reply:
Made change, let me know if you disagree and we can rollback